### DISCUSSION TITLE
ビルド中 webpack  で Error: Unknown font format が出て Material Icons 等が表示されない

### DISCUSSION THREAD

**Discussion by yasumichi** [growilabs/growi#8779] [Help / Questions] [answered]
Ubuntu Server 22.04 LTS 上に GROWI をデプロイしようとしています。

初回、`npm start` でのビルド時に webpack が以下のようなエラーを出力します。 

```
@growi/app:build: Failed to load font file: /opt/growi/apps/app/resource/fonts/MaterialSymbolsOutlined-opsz,wght,FILL@20..48,300,0..1.	
@growi/app:build: Error: Unknown font format
@growi/app:build: Failed to load font file: /opt/growi/apps/app/resource/fonts/SourceHanCodeJP-Regular-subset-main.woff2
@growi/app:build: Error: Unknown font format
@growi/app:build: Failed to load font file: /opt/growi/apps/app/resource/fonts/SourceHanCodeJP-Regular-subset-jis2.woff2
@growi/app:build: Error: Unknown font format
@growi/app:build: Failed to load font file: /opt/growi/apps/app/resource/fonts/PressStart2P-latin.woff2
@growi/app:build: Error: Unknown font format
```

その結果、以下のように Material Icons が表示されませんでした。

![スクリーンショット 2024-05-01 072255](https://github.com/weseek/growi/assets/802334/7fe75964-3aab-4899-89d9-bf53b18cefa5)

何か、考えられる原因等ありますでしょうか？

**Comment by yuki-takei** [comment, marked as answer]
# ビルド手順について
現時点の [weseek/growi](https://github.com/weseek/growi) のリポジトリは npm ではなく yarn または turbo コマンドでの運用を想定しています。
もし [Ubuntu Server](https://docs.growi.org/ja/admin-guide/getting-started/ubuntu-server.html) のドキュメントを見て進められたのであればこちらは古いので注意してください(アップデートされていなくて申し訳ないです)

# バイナリファイルについて
エラーログから Material Symbols だけでなくSource Han Code JP 等も見つかっていないようなので、おそらく Git LFS 管理のファイルを取得できていないものと思われます。`git lfs pull` を試してみてください。

**Reply by yasumichi** [reply]
ありがとうございました。

`git lfs pull` で症状が改善しました。

なるほど、ファイルが存在しているように見えて、メタデータのような参照の情報が入っている小さなファイルが出来ていたのですね。

- [Git LFS のリポジトリをcloneやpullしたときにどうなるのか #Git - Qiita](https://qiita.com/yuki2006/items/6893d7fcc1014c5cfb41)

> Git LFSを導入してなくても clone/pullをすることができ、しかもエラーになりません。
>
> 当該のファイルは メタデータのような参照の情報が入っている小さなファイルになっています。
> ビルド時に初めて気づくことになったり、はたまた本番運用中に気づくということにもなりかねません。
> なので周知力が必要です。

なお、yarn および turbo については、インストールしていました。

現在、[Update for Ubuntu Server 22.04 LTS. #459](https://github.com/weseek/growi-docs/pull/459)にて、Ubuntu Server でのインストール手順について、Pull request を上げていますが、

```
$ sudo apt install git-lfs
```

の手順が必要ですね。

**Reply by yasumichi** [reply]
[Ubuntu Server | GROWI Docs](https://docs.growi.org/ja/admin-guide/getting-started/ubuntu-server.html#%E6%A6%82%E8%A6%81) の修正として、`git-lfs` のインストールについて、追記し、再度、Pull request を上げなおしました。

[Update for Ubuntu Server 22.04 LTS by yasumichi · Pull Request #460 · weseek/growi-docs](https://github.com/weseek/growi-docs/pull/460)
